src/extension/wps/src/main/java/au/org/emii/wps/GoGoDuckProcess.java

```diff
@@ -54,29 +67,65 @@ public FileRawData execute(
             ProgressListener progressListener
     ) throws ProcessException {
         try {
-            if (goGoDuckConfig.getThreadCount() <= 0) {
-                throw new ProcessException("threadCount set to 0 or below, job will not run");
-            }
-
-            File outputFile;
+            Path convertedFile;
+            String mimeType;
+            String extension;
 
             try {
-                outputFile = getResourceManager().getTemporaryResource(".nc").file();
-            } catch (Exception e) {
-                logger.error(e.toString(), e);
-                throw new GoGoDuckException("Unable to obtain temporary file required for aggregation", e);
-            }
+                SubsetParameters parameters = SubsetParameters.parse(subset);
+
+                IndexReader indexReader = new FeatureSourceIndexReader(catalog, config.getUrlSubstitution(layer));
+
+                Set<DownloadRequest> downloads = indexReader.getDownloadList(layer, config.getTimeField(),
+                    config.getSizeField(), config.getFileUrlField(), parameters);
+
+                enforceFileLimits(downloads, config.getFileLimit(), config.getFileSizeLimit());
+
+                Path workingDir = Paths.get(getWorkingDir());
+                Path downloadDir = workingDir.resolve("downloads");
+                Files.createDirectories(downloadDir);
 
-            String filePath = outputFile.toPath().toAbsolutePath().toString();
+                DownloadConfig downloadConfig = new DownloadConfig.ConfigBuilder()
+                    .downloadDirectory(downloadDir)
+                    .localStorageLimit(200 * 1024 * 1024)
+                    .build();
 
-            GoGoDuck ggd = new GoGoDuck(catalog, layer, subset, filePath, format, goGoDuckConfig);
+                Downloader downloader = new Downloader(60 * 1000, 60 * 1000); //TODO: allow settings to be configured
+                ExecutorService pool = Executors.newFixedThreadPool(8); //TODO: and here
```

Comment:
Not sure why I exposed this - yes it can probably become an implementation detail for the DownloadManager.  External configuration would be added in the config task I'm working on.